test/extensions/filters/http/cache/cache_filter_test.cc

```diff
@@ -608,22 +647,239 @@ TEST_F(CacheFilterTest, FilterDeletedBeforePostedCallbackExecuted) {
   }
 }
 
-// A new type alias for a different type of tests that use the exact same class
+// A new type alias for a different type of tests that use the exact same class.
+// In these tests, realistically the data in request 1 should be encoded in several chunks too,
+// however, the only purpose of request 1 is to put the response in the cache, so it shouldn't
+// matter.
+// Cases where the body size is less than the buffer_limit_ are not exercised as they are
+// already tested in the above tests.
+using CacheChunkSizeTest = CacheFilterTest;
+
+// Test that a body with size exactly equal to the buffer limit will be encoded in 1 chunk.
+TEST_F(CacheChunkSizeTest, EqualBufferLimit) {
+  request_headers_.setHost("EqualBufferLimit");
+  const std::string body = std::string(buffer_limit_, 'a');
```

Comment:
again I'd consider using smaller limits and more interesting strings to compare.

Reply:
Done.